dynamo-insert-actions/dynamo-insert-actions.js

```diff
@@ -0,0 +1,283 @@
+    try {
+        awsClient._initSTS();
+        await awsClient._getCallerIdentity();
+    } catch (error) {
+        if (error.name === 'CredentialsProviderError' ||
+            error.message?.includes('expired') ||
+            error.message?.includes('credentials')) {
+            console.error('\n=== SSO Authentication Error ===');
+            console.error('Your SSO session has expired or is invalid.');
+            console.error('Please run the following commands:');
+            console.error('1. aws sso logout');
+            console.error(`2. aws sso login --profile ${awsClient.ssoProfile}`);
+            process.exit(1);
+        }
+        throw error;
+    }
+}
+
+/**
+ * Prints a summary of the execution results
+ * @param {Object} stats - Statistics object containing results
+ * @param {number} stats.totalProcessed - Total number of items processed
+ * @param {number} stats.successCount - Number of successful updates
+ * @param {number} stats.failureCount - Number of failed updates
+ * @param {number} stats.unprocessedCount - Number of unprocessed items
+ */
+function printSummary(stats) {
+    console.log('\n=== Execution Summary ===');
+    console.log(`\nTotal items processed: ${stats.totalProcessed}`);
+    console.log(`Successfully updated: ${stats.successCount}`);
+    console.log(`Failed updates: ${stats.failureCount}`);
+    if (stats.unprocessedCount > 0) {
+        console.log(`Unprocessed items: ${stats.unprocessedCount}`);
+    }
+    console.log('\nResults written to:');
+    console.log('- results/failure.json');
+}
+
+/**
+ * Main execution function
+ * Processes CSV file and updates DynamoDB items with new TTL values
+ * and sets notToHandle flag to true
+ * @throws {Error} If AWS operations fail or file operations fail
+ */
+async function main() {
+    const args = validateArgs();
+    const { envName, csvFile, ttlDays, actionId } = args;
+
+    initializeResultsFiles();
+
+    const coreClient = new AwsClientsWrapper('core', envName);
+    
+    await testSsoCredentials(coreClient);
+    
+    coreClient._initDynamoDB();
+
+    const records = parseCsvFile(csvFile, actionId);
+    const itemsToInsert = prepareItemsForInsert(records, ttlDays);
+
+    console.log(`Processing ${itemsToInsert.length} items to insert...`);
+    
+    // Perform batch writes with strict error handling
+    for (let i = 0; i < itemsToInsert.length; i += 25) {
+        const batch = itemsToInsert.slice(i, i + 25);
+        const result = await coreClient._batchWriteItem('pn-Action', batch);
+        if (result.UnprocessedItems && Object.keys(result.UnprocessedItems).length > 0) {
+            console.error('Failed to process all items in batch');
+            process.exit(1);
+        }
+        console.log(`Inserted items ${i + 1} to ${Math.min(i + 25, itemsToInsert.length)}`);
+    }
+
+    // Verify inserts (will exit on first failure)
+    const failures = await verifyInserts(coreClient, itemsToInsert);
```

Comment:
@elionello-ext Non mi sembra necessario, il task richiedeva;

`ATTENZIONE: in questo caso è necessario verificare l’output della richiesta batch verso dynamo per capire se tutti gli input sono stati aggiornati.`

Per recuperare le failure devi interpretare il Risultato della batch write